### PR TITLE
Upgrade node

### PR DESCRIPTION
**Pull request by snowystinger**
Closes 

Notes for after merging:
Everyone will need to `make clean_all`, switch to node 18, then `yarn` so that all packages are rebuilt against the new node.

Why this has webpack patches:
Held off on SB7 update until https://github.com/parcel-bundler/parcel/issues/8668 is resolved.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

**Comment by snowystinger** [comment]
`export NODE_OPTIONS=--openssl-legacy-provider` does not get this working
nor does putting it into the script targets, nor making a make target and putting it in that

Using the patch-package I've come up with here, does work

I suspect we'll need something similar in our `examples` which use webpack4

File: patches/webpack+4.46.0.patch [hunk 1]
```diff
@@ -0,0 +1,144 @@
+diff --git a/node_modules/webpack/lib/Compilation.js b/node_modules/webpack/lib/Compilation.js
+index 6329174..196c4bb 100644
+--- a/node_modules/webpack/lib/Compilation.js
++++ b/node_modules/webpack/lib/Compilation.js
+@@ -1915,7 +1915,7 @@ class Compilation extends Tapable {
+ 		const hashFunction = outputOptions.hashFunction;
+ 		const hashDigest = outputOptions.hashDigest;
+ 		const hashDigestLength = outputOptions.hashDigestLength;
+-		const hash = createHash(hashFunction);
++		const hash = createHash('sha512');
```
Comment (devongovett) [inline patches/webpack+4.46.0.patch:10]: Is there a way to set the `hashFunction` in the webpack config instead of overriding the source code?

Comment (snowystinger) [reply patches/webpack+4.46.0.patch:10]: https://github.com/adobe/react-spectrum/pull/4945#issuecomment-1692721796
> Using the patch-package I've come up with here, does work. I derived this from the later comment in this StackOverflow https://stackoverflow.com/questions/69692842/error-message-error0308010cdigital-envelope-routinesunsupported where it's titled "Danger". I tried just passing the new hashing algorithm via the webpack config, however, it didn't appear to do anything, and there are several places where the wrong hash algorithm is hardcoded.

**Review by reidbarber — APPROVED** [review]

File: packages/@react-spectrum/datepicker/test/DatePicker.test.js [hunk 2]
```diff
@@ -33,7 +33,7 @@ function getTextValue(el) {
 }
 
 function expectPlaceholder(el, placeholder) {
-  expect(getTextValue(el)).toBe(placeholder);
+  expect(getTextValue(el).replace(' ', ' ')).toBe(placeholder);
```
Comment (LFDanLu) [inline packages/@react-spectrum/datepicker/test/DatePicker.test.js:36]: What exactly is this replace for?

File: packages/@react-spectrum/datepicker/test/DatePicker.test.js
Comment (LFDanLu) [inline packages/@react-spectrum/datepicker/test/DatePicker.test.js:1]: What are all the whitespace changes in these test files for? I see some `no-irregular-whitespace` eslint ignores, is it related to that

File: packages/dev/storybook-builder-parcel/index.js [hunk 3]
```diff
@@ -62,6 +62,8 @@ exports.build = async function ({options}) {
 exports.corePresets = [];
 exports.previewPresets = [];
 
+exports.bail = async function (e) {};
```
Comment (LFDanLu) [inline packages/dev/storybook-builder-parcel/index.js:65]: What is this for?

**Review by LFDanLu — COMMENTED** [review]
Posted some questions, still looking for the publish stats for the webpack 4 build on verdaccio

**Comment by LFDanLu** [comment]
For other reviewers, here are the webpack 4 example app links:
https://reactspectrum.blob.core.windows.net/reactspectrum/305000df465b136b5852f6160f9b4a1f3e080d72/verdaccio/publish-stats/webpack-4-build-stats.txt
https://reactspectrum.blob.core.windows.net/reactspectrum/305000df465b136b5852f6160f9b4a1f3e080d72/verdaccio/webpack-4/bundle.js

File: packages/@react-spectrum/datepicker/test/DatePicker.test.js
Comment (snowystinger) [reply packages/@react-spectrum/datepicker/test/DatePicker.test.js:1]: yeah, it's all due to this, the comment was just further down in the diff but that was the first place I ran into it in the code

https://github.com/adobe/react-spectrum/pull/4945/files#diff-d1618c3d21a2ae1b90bc6c86a9c52956765b70ac780b4edf60e846a0465e2556R642

File: packages/@react-spectrum/datepicker/test/DatePicker.test.js (hunk 2 above)
Comment (snowystinger) [reply packages/@react-spectrum/datepicker/test/DatePicker.test.js:36]: see https://github.com/adobe/react-spectrum/pull/4945/files#r1322233532

File: packages/dev/storybook-builder-parcel/index.js (hunk 3 above)
Comment (snowystinger) [reply packages/dev/storybook-builder-parcel/index.js:65]: ah, that slipped by me
shouldln't hurt anything, but it's required when we go to upgrade to sb7

**Review by LFDanLu — APPROVED** [review]

**Pull request merged**